src/probability/independence.lean

```diff
@@ -124,6 +127,42 @@ def indep_fun {α β γ} [measurable_space α] [mβ : measurable_space β] [mγ
   (f : α → β) (g : α → γ) (μ : measure α . volume_tac) : Prop :=
 indep (measurable_space.comap f mβ) (measurable_space.comap g mγ) μ
 
+/-- Conditional version of `Indep_sets` that takes a set of conditioning events `C` and requires
+`∀ (c ∈ C), Indep_sets π μ[|c]`. -/
+def cond_Indep_sets {α ι} [measurable_space α] (π : ι → set (set α))
+  (C : set (set α)) (μ : measure α . volume_tac) : Prop :=
+∀ (c ∈ C), Indep_sets π $ μ[|c]
+
+/-- Conditional version of `indep_sets` that takes a set of conditioning events `C` and requires
+`∀ c ∈ C, indep_sets s1 s2 μ[|c]`. -/
+def cond_indep_sets {α} [measurable_space α] (s1 s2 : set (set α)) (C : set (set α))
+  (μ : measure α . volume_tac) : Prop :=
+∀ (c ∈ C), indep_sets s1 s2 $ μ[|c]
+
+/-- Conditional version of `Indep` that takes a conditioning measurable space `C` and requires
+`∀ c ∈ C, Indep m μ[|c]`. -/
+def cond_Indep {α ι} (m : ι → measurable_space α) (C : measurable_space α)
+  [measurable_space α] (μ : measure α . volume_tac) : Prop :=
+cond_Indep_sets (λ x, (m x).measurable_set') C.measurable_set' μ 
+
+/-- Conditional version of `indep` that takes a conditioning measurable space `C` and requires
+`∀ c ∈ C, indep s1 s2 μ[|c]`. -/
+def cond_indep {α} (m₁ m₂ : measurable_space α) (C : measurable_space α)
+  [measurable_space α] (μ : measure α . volume_tac) : Prop :=
+cond_indep_sets (m₁.measurable_set') (m₂.measurable_set') C.measurable_set' μ
```

Comment:
`m₁.measurable_set'` is not a set, so this invocation doesn't quite make sense. @JasonKYi, can you remember what we decided the preferred spelling was?

Reply:
I think we decided to use `{s | measurable_set[m₁] s}`

Reply:
```suggestion
cond_indep_sets {x | measurable_set[m₁] x} {x | measurable_set[m₂] x} {x | measurable_set[C] x} μ
```